src/main/java/gift/validation/validator/KakaoApprovalValidator.java

```diff
@@ -0,0 +1,32 @@
+package gift.validation.validator;
+
+import gift.validation.annotation.KakaoApprovalRequired;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import org.springframework.beans.BeanWrapperImpl;
+
+public class KakaoApprovalValidator implements ConstraintValidator<KakaoApprovalRequired, Object> {
+
+    private String nameField;
+    private String approvalField;
+
+    @Override
+    public void initialize(KakaoApprovalRequired constraintAnnotation) {
+        this.nameField = constraintAnnotation.nameField();
+        this.approvalField = constraintAnnotation.approvalField();
+    }
+
+    @Override
+    public boolean isValid(Object value, ConstraintValidatorContext context) {
+        if (value == null) return true;
+
+        BeanWrapperImpl wrapper = new BeanWrapperImpl(value);
+        String name = (String) wrapper.getPropertyValue(nameField);
+        Boolean isMdApproved = (Boolean) wrapper.getPropertyValue(approvalField);
+
+        if (name == null || !name.contains("카카오")) {
+            return true;
+        }
```

Comment:
```suggestion
        if (name == null || !name.contains("카카오")) {
            return true;
        }
        Boolean isMdApproved = (Boolean) wrapper.getPropertyValue(approvalField);
```
순서가 바뀌면 좋을 것 같아요. name 에 카카오가 없으면 isMdApproved를 굳이 검사하지 않아도 될 것 같아서요 :)